common/components/deploy_model/spec.yaml

```diff
@@ -0,0 +1,77 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+name: deploy_model
+version: 0.0.1
+type: command
+
+is_deterministic: True
+
+display_name: Deploy model
+description:
+  Deploy a model to a workspace. The component works on compute with [MSI](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-create-manage-compute-instance?tabs=python) attached.
+
+environment: azureml:python-sdk-v2:1
+
+code: ../../src/model_deployment
+command: >-
+  python deploy.py
+  --registration_details ${{inputs.registration_details}}
+  --endpoint_name ${{inputs.endpoint_name}}
+  --deployment_name ${{inputs.deployment_name}}
+  --instance_type ${{inputs.instance_type}}
+  --instance_count ${{inputs.instance_count}}
+  $[[--max_concurrent_requests_per_instance ${{inputs.max_concurrent_requests_per_instance}}]] 
+  $[[--request_timeout_ms ${{inputs.request_timeout_ms}}]]
+  $[[--max_queue_wait_ms ${{inputs.max_queue_wait_ms}}]]
+  --model_deployment_details ${{outputs.model_deployment_details}}
+
+inputs:
+  # Output of registering component
+  registration_details:
+    type: uri_file
+    optional: false
+    description: Text file that contains the ID of registered model to be deployed
```

Comment:
nit: Update to: `File that contains model registration details`?

Reply:
Any specific reason for this ? Last time louie specifically mentioned to add file type (like text,json) to have clarity

Reply:
Yes, specifying a file type is needed.
What I was thinking is we probably need to dump more details like:
```
model_name: <>
model_version: <>
model_type: <>
model_flavors: <>
id: <> [Asset URI]
path: <> [datastore/storage asset URI]
registry or workspace registration? If registry have that as well [This can be figured out from id as well, but having an explicit property will not do any harm]
```

The reason having these details would be useful is, though there is a way to figure out some details using asset ID itself, but for some other we would need to do models.get operation which would require downstream components to do n/w call as well as run on MSI compute. Please let me know what you think and whether having these details make sense in registration component which can be consumed in deploy component

Reply:
Make sense @novaturient95 .We will need model_type parameter, when we will add support deployment for custom model  but in current case we do not require such parameter.
Shall we add this parameters in reg component in next release,which will include support for custom model?

Reply:
yeah, as part of this PR or next PR is okay